### DISCUSSION TITLE
Rate limit reached frequently

### DISCUSSION THREAD

**Discussion by janedenone** [mastodon/mastodon#19661] [Q&A]
Hi,

when accessing Mastodon (private instance) from more than one device (or sometimes even with just one device), I frequently hit the rate limit for GET requests. This problem was already described with a [suggestion](https://github.com/mastodon/mastodon/issues/15832) to solve it with configurable limits, but I wonder if it can be solved without touching the code. I imagine that this affects many users, irrespective of the capacity of their Mastodon instance.

Thanks,
Jan

**Comment by phocks** [comment]
> I think this might be the file that controls some rate limits. I'm wondering if I bump these numbers on my single-user instance and then re-compile, if that would do the trick. I'm basically trying to loop through my following list and make changes. Currently I'm just using an async `sleep` function to throttle requests, but because it's my own server, I feel like I should be able to raise the rate limit myself. https://github.com/mastodon/mastodon/blob/46ad7fea9d67631f54dd1ef45114a08cd2c5db73/config/initializers/rack_attack.rb

I just replied here. Hopefully it's helpful to someone https://github.com/mastodon/mastodon/issues/15832#issuecomment-1312563393

**Reply by phocks** [reply]
Can confirm, raising this number from 300 to 99999 raised the api limit for me.
https://github.com/mastodon/mastodon/blob/46ad7fea9d67631f54dd1ef45114a08cd2c5db73/config/initializers/rack_attack.rb#L49

**Reply by janedenone** [reply]
> > I think this might be the file that controls some rate limits. I'm wondering if I bump these numbers on my single-user instance and then re-compile, if that would do the trick. I'm basically trying to loop through my following list and make changes. Currently I'm just using an async `sleep` function to throttle requests, but because it's my own server, I feel like I should be able to raise the rate limit myself. https://github.com/mastodon/mastodon/blob/46ad7fea9d67631f54dd1ef45114a08cd2c5db73/config/initializers/rack_attack.rb
> 
> I just replied here. Hopefully it's helpful to someone [#15832 (comment)](https://github.com/mastodon/mastodon/issues/15832#issuecomment-1312563393)

Thanks! After changing `rake_attack.rb`, executing `bundle exec rails assets:precompile` gives me:

```
yarn install v1.22.19
[1/6] Validating package.json...
[2/6] Resolving packages...
success Already up-to-date.
Done in 1.02s.
Everything's up-to-date. Nothing to do
```

What am I missing?

**Reply by phocks** [reply]
I'm not sure if the re-compiling is actually necessary now that I think of it. Ruby will just interpret the code in the file. You may need to restart the Mastodon services as described in this page https://docs.joinmastodon.org/admin/upgrading/

**Comment by nfriedly** [comment]
Any suggestions for someone using the docker image? (Editing the source code is probably possible, but not nearly as straight forward.)

**Comment by roke-julian-lockhart** [comment]
> https://github.com/mastodon/mastodon/discussions/19661#discussion-4529745

I was rate limited because I have multiple tabs open, as https://www.reddit.com/r/Mastodon/comments/10mlg09/comment/j69nvw1/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button elaborates.

**Comment by ghost** [comment]
I am self-hosting a Mastodon server and I am hitting these rate limits suddenly with just a single tab open. I really need to know how to raise these limits as it's making Mastodon painful to use.

**Comment by flancian** [comment]
See also:

https://github.com/mastodon/mastodon/issues/15832
https://github.com/mastodon/mastodon/issues/22217

And probably several others which have been open for 3-4 years respectively. Maybe they are not getting triaged because they currently are classified as "suggestions" -- but the impact this has on users makes me think they should be "bugs" instead?
